webapp/src/components/common/Form/index.tsx

```diff
@@ -123,14 +131,17 @@ function Form<TFieldValues extends FieldValues, TContext>(
       : config?.defaultValues,
   });
 
-  const { getValues, setValue, handleSubmit, formState, reset } = formApi;
+  const { getValues, setValue, setError, handleSubmit, formState, reset } =
+    formApi;
   // * /!\ `formState` is a proxy
-  const { isSubmitting, isSubmitSuccessful, isDirty, dirtyFields } = formState;
+  const { isSubmitting, isSubmitSuccessful, isDirty, dirtyFields, errors } =
+    formState;
   // Don't add `isValid` because we need to trigger fields validation.
   // In case we have invalid default value for example.
   const isSubmitAllowed = isDirty && !isSubmitting;
   const showSubmitButton = !hideSubmitButton && !autoSubmitConfig.enable;
   const showFooter = showSubmitButton || enableUndoRedo;
+  const rootError = errors.root?.[ROOT_ERROR_KEY];
```

Comment:
maybe you can explain a bit more about the usage of `default` key. If I understand correctly, this allows errors to be persisted and keep any errors that have not yet been corrected across submits?

Reply:
I will go into more detail